packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js

```diff
@@ -201,6 +219,144 @@ const handleEnterKeyOnList = (editor) => {
   }
 };
 
+const Link = React.forwardRef(({ element, children, ...attributes }, forwardedRef) => {
+  const { formatMessage } = useIntl();
+  const editor = useSlate();
+  const path = ReactEditor.findPath(editor, element);
+  const [popoverOpen, setPopoverOpen] = React.useState(
+    editor.lastInsertedLinkPath ? Path.equals(path, editor.lastInsertedLinkPath) : false
+  );
+  const [isEditing, setIsEditing] = React.useState(element.url === '');
+  const linkRef = React.useRef(null);
+
+  const elementText = element.children.map((child) => child.text).join('');
+
+  const handleOpenEditPopover = (e) => {
+    e.preventDefault();
+    setPopoverOpen(true);
+  };
+
+  const handleSave = (e) => {
+    e.stopPropagation();
+    const { text, url } = e.target.elements;
+
+    // If the selection is collapsed, we select the parent node because we want all the link to be replaced
+    if (Range.isCollapsed(editor.selection)) {
+      const [, parentPath] = Editor.parent(editor, editor.selection.focus?.path);
+      Transforms.select(editor, parentPath);
+    }
+
+    editLink(editor, { url: url.value, text: text.value });
+    setIsEditing(false);
+    setPopoverOpen(false);
+  };
+
+  const handleDismiss = () => {
+    setPopoverOpen(false);
+    ReactEditor.focus(editor);
+  };
+
+  const composedRefs = composeRefs(linkRef, forwardedRef);
+
+  return (
+    <>
+      <BaseLink
+        {...attributes}
+        ref={composedRefs}
+        href={element.url}
+        onClick={handleOpenEditPopover}
+      >
+        {children}
+      </BaseLink>
+      {popoverOpen && (
+        <Popover
+          source={linkRef}
+          onDismiss={handleDismiss}
+          padding={4}
+          placement="right-end"
+          contentEditable={false}
+          onEscapeKeyDown={() => ReactEditor.focus(editor)}
```

Comment:
is `onDismiss` not called when we press Escape? Just wondering why we're doubling down on focus